src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Resources/Get/GetResourceHandlerTests.cs

```diff
@@ -0,0 +1,238 @@
+        var patient = Samples.GetDefaultPatient();
+        var rawResource = new RawResource(patient.ToJson(), FhirResourceFormat.Json, false);
+        var wrapper = new ResourceWrapper(
+            patient,
+            rawResource,
+            new ResourceRequest(System.Net.Http.HttpMethod.Get),
+            false,
+            null,
+            null,
+            null);
+
+        _fhirDataStore.GetAsync(Arg.Any<ResourceKey>(), Arg.Any<CancellationToken>())
+            .Returns(wrapper);
+
+        // Setup context accessor default behavior
+        _contextAccessor.RequestContext.Returns(Substitute.For<IFhirRequestContext>());
+        _contextAccessor.RequestContext.AccessControlContext.Returns((AccessControlContext)null);
+    }
+
+    [Fact]
+    public async Task GivenAGetResourceRequest_WhenUserHasReadPermission_ThenGetShouldSucceed()
+    {
+        // Arrange
+        var authService = Substitute.For<IAuthorizationService<DataActions>>();
+        var getResourceHandler = new GetResourceHandler(
+            _fhirDataStore,
+            _conformanceProvider,
+            _resourceWrapperFactory,
+            _resourceIdProvider,
+            _dataResourceFilter,
+            authService,
+            _contextAccessor,
+            _searchService);
+
+        authService
+            .CheckAccess(DataActions.Read | DataActions.ReadV2, CancellationToken.None)
+            .Returns(DataActions.Read);
+
+        var request = new GetResourceRequest(new ResourceKey("Patient", "123"), bundleResourceContext: null);
+
+        // Act & Assert - Should not throw UnauthorizedFhirActionException
+        var result = await getResourceHandler.Handle(request, CancellationToken.None);
+
+        Assert.NotNull(result);
+        Assert.NotNull(result.Resource);
+    }
+
+    [Fact]
+    public async Task GivenAGetResourceRequest_WhenUserHasReadV2Permission_ThenGetShouldSucceed()
+    {
+        // Arrange
+        var authService = Substitute.For<IAuthorizationService<DataActions>>();
+        var getResourceHandler = new GetResourceHandler(
+            _fhirDataStore,
+            _conformanceProvider,
+            _resourceWrapperFactory,
+            _resourceIdProvider,
+            _dataResourceFilter,
+            authService,
+            _contextAccessor,
+            _searchService);
+
+        authService
+            .CheckAccess(DataActions.Read | DataActions.ReadV2, CancellationToken.None)
+            .Returns(DataActions.ReadV2);
+
+        var request = new GetResourceRequest(new ResourceKey("Patient", "123"), bundleResourceContext: null);
+
+        // Act & Assert - Should not throw UnauthorizedFhirActionException
+        var result = await getResourceHandler.Handle(request, CancellationToken.None);
+
+        Assert.NotNull(result);
+        Assert.NotNull(result.Resource);
+    }
+
+    [Fact]
+    public async Task GivenAGetResourceRequest_WhenUserLacksPermissions_ThenUnauthorizedExceptionIsThrown()
```

Comment:
Just a suggestion - should we club all the UnauthorizedFhirActionException facts together into a theory?